commit 035f4769fec028745773bd08f4a75784b84f2e6c
Author: Thomas Debesse <dev@illwieckz.net>
Date:   Fri Oct 21 00:39:52 2022 +0200

    patch: luxmark3/luxmark3-luxmark-lowercase-false.patch

diff --git a/src/mainwindow.cpp b/src/mainwindow.cpp
index 8e4d75c..7eefed4 100644
--- a/src/mainwindow.cpp
+++ b/src/mainwindow.cpp
@@ -379,7 +379,7 @@ void MainWindow::SetHardwareTreeModel(HardwareTreeModel *treeModel) {
 }
 
 bool MainWindow::event(QEvent *event) {
-	bool retval = FALSE;
+	bool retval = false;
 	int eventtype = event->type();
 
 	// Check if it's one of "our" events
