commit bcfc72ee40bb5d9f0f6f1a9532c0f8b1510257af
Author: Thomas Debesse <dev@illwieckz.net>
Date:   Mon Oct 24 10:25:32 2022 +0200

    patch: luxmark3/luxmark3-luxcore-link-slg4-with-opencl.patch

diff --git a/samples/smallluxgpu4/CMakeLists.txt b/samples/smallluxgpu4/CMakeLists.txt
index 2b6d659a2..599671e70 100644
--- a/samples/smallluxgpu4/CMakeLists.txt
+++ b/samples/smallluxgpu4/CMakeLists.txt
@@ -30,9 +30,9 @@ set(SMALLLUXGPU_SRCS
 add_executable(slg4 ${SMALLLUXGPU_SRCS})
 
 if(APPLE)
-	TARGET_LINK_LIBRARIES(slg4 luxcore smallluxgpu luxrays ${EMBREE_LIBRARY} ${OPENGL_LIBRARIES} ${GLEW_LIBRARY} ${GLUT_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+	TARGET_LINK_LIBRARIES(slg4 luxcore smallluxgpu luxrays ${EMBREE_LIBRARY} ${OPENGL_LIBRARIES} ${OpenCL_LIBRARIES} ${GLEW_LIBRARY} ${GLUT_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
 else(APPLE)
-	TARGET_LINK_LIBRARIES(slg4 luxcore smallluxgpu luxrays ${EMBREE_LIBRARY} ${OPENGL_LIBRARIES} ${GLEW_LIBRARY} ${GLUT_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+	TARGET_LINK_LIBRARIES(slg4 luxcore smallluxgpu luxrays ${EMBREE_LIBRARY} ${OPENGL_LIBRARIES} ${OpenCL_LIBRARIES} ${GLEW_LIBRARY} ${GLUT_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
 endif(APPLE)
 
 # This instructs FREEGLUT to emit a pragma for the static version
