diff --git a/src/luxrays/idevices/openclidevice.cpp b/src/luxrays/idevices/openclidevice.cpp
index b8b23bbf3..93c2a6ef8 100644
--- a/src/luxrays/idevices/openclidevice.cpp
+++ b/src/luxrays/idevices/openclidevice.cpp
@@ -280,6 +280,10 @@ void OpenCLIntersectionDevice::Start() {
 
 	kernels = nullptr;
 
+	const char* env_disable_image = getenv("LUXCORE_DISABLE_IMAGE");
+	const bool disable_image = (env_disable_image != nullptr && strlen(env_disable_image) != 0);
+	enableImageStorage = (enableImageStorage && !disable_image);
+
 	oclQueues.clear();
 	if (dataParallelSupport) {
 		// Compile all required kernels
